README.md

```diff
@@ -30,19 +31,27 @@ $encryptor = ObjectEncryptorFactory::getEncryptor(
-- `ProjectAKVWrapper` - current Azure wrapper for `KBC::ProjectSecureKV::` ciphers, requires `akvUrl`, `stackId`, `componentId` and `projectId`.
-- `ConfigurationAKVWrapper` - current Azure wrapper for `KBC::ConfigSecureKV::` ciphers, requires `akvUrl`, `stackId`, `componentId`, `projectId` and `configurationId`.
+- `GenericKMSWrapper` - AWS wrapper for `KBC::Secure::` ciphers, requires `kmsKeyId` and `kmsRegion`. Also, the runner must have AWS credentials available (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`)
+- `ComponentKMSWrapper` - AWS wrapper for `KBC::ComponentSecure::` ciphers, requires `kmsKeyId`, `kmsRegion`, `stackId` and `componentId`
+- `ProjectKMSWrapper` - AWS wrapper for `KBC::ProjectSecure::` ciphers, requires `kmsKeyId`, `kmsRegion`, `stackId`, `componentId` and `projectId`.
+- `ConfigurationKMSWrapper` - AWS wrapper for `KBC::ConfigSecure::` ciphers, requires `kmsKeyId`, `kmsRegion`, `stackId`, `componentId`, `projectId` and `configurationId`.
+- `ProjectWideKMSWrapper` - KMS wrapper `KBC::ProjectWideSecure::` ciphers, requires `kmsKeyId`, `kmsRegion`, `stackId`, and `projectId`.
+- `BranchTypeProjectKMSWrapper` - KMS wrapper `KBC::BranchTypeSecure::` ciphers, requires `kmsKeyId`, `kmsRegion`, `stackId`, `componentId`, projectId` and `branchType`.
+- `BranchTypeConfigurationKMSWrapper` - KMS wrapper `KBC::BranchTypeConfigSecure::` ciphers, requires `kmsKeyId`, `kmsRegion`, `stackId`, `componentId`, `projectId`, `configurationId` and `branchType`.
+- `BranchTypeProjectWideKMSWrapper` - KMS wrapper `KBC::ProjectWideBranchTypeSecure::` ciphers, requires `kmsKeyId`, `kmsRegion`, `stackId`, `projectId` and `branchType`.
+- `GenericAKVWrapper` - Azure wrapper for `KBC::SecureKV::` ciphers, requires `akvUrl`. Also, the runner must have AWS credentials available (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`)
+- `ComponentAKVWrapper` - Azure wrapper for `KBC::ComponentSecureKV::` ciphers, requires `akvUrl`, `stackId` and `componentId`
+- `ProjectAKVWrapper` - Azure wrapper for `KBC::ProjectSecureKV::` ciphers, requires `akvUrl`, `stackId`, `componentId` and `projectId`.
+- `ConfigurationAKVWrapper` - Azure wrapper for `KBC::ConfigSecureKV::` ciphers, requires `akvUrl`, `stackId`, `componentId`, `projectId` and `configurationId`.
+- `ProjectWideAKvWrapper` - Azure wrapper `KBC::ProjectWideSecureKV::` ciphers, requires `akvUrl`, `stackId`, and `projectId`.
+- `BranchTypeProjectAKVWrapper` - Azure wrapper `KBC::BranchTypeSecureKV::` ciphers, requires `akvUrl`, `stackId`, `componentId`, projectId` and `branchType`.
+- `BranchTypeConfigurationAKVWrapper` - Azure wrapper `KBC::BranchTypeConfigSecureKV::` ciphers, requires `akvUrl`, `stackId`, `componentId`, `projectId`, `configurationId` and `branchType`.
+- `BranchTypeProjectWideAKVWrapper` - Azure wrapper `KBC::ProjectWideBranchTypeSecureKV::` ciphers, requires `akvUrl`, `stackId`, `projectId` and `branchType`.

```

Comment:
Potrebujeme to tu faklt vyjmenovavat? :smile: Nejak mi otika uzitecnost toho

Reply:
jo, až budeš někdy potřebovat vědět, jestli BranchTypeConfigurationAKVWrapper potřebuje componentId nebo ne, tak mě budeš děkovat

Reply:
Kouknu do kodu :wink:

Reply:
Tak určitě:

https://github.com/keboola/object-encryptor/blob/90bdd50cdd065d22b08224def3fd6d8e588fb940/src/Wrapper/BranchTypeConfigurationAKVWrapper.php#L23-L25